core/planner.cpp

```diff
@@ -656,22 +649,17 @@ std::vector<decostop> plan(struct deco_state *ds, struct diveplan &diveplan, str
 	create_dive_from_plan(diveplan, dive, dc, is_planner);
 
 	// Do we want deco stop array in metres or feet?
-	if (prefs.units.length == units::METERS ) {
-		decostoplevels = decostoplevels_metric;
-		decostoplevelcount = std::size(decostoplevels_metric);
-	} else {
-		decostoplevels = decostoplevels_imperial;
-		decostoplevelcount = std::size(decostoplevels_imperial);
-	}
+	auto &decostoplevels = prefs.units.length == units::METERS ?
+		decostoplevels_metric : decostoplevels_imperial;
 
 	/* If the user has selected last stop to be at 6m/20', we need to get rid of the 3m/10' stop.
 	 * Otherwise reinstate the last stop 3m/10' stop.
 	 * Remark: not reentrant, but the user probably won't change preferences while this is running.
 	 */
 	if (prefs.last_stop)
-		*(decostoplevels + 1) = 0;
+		decostoplevels[1] = 0_m;
 	else
-		*(decostoplevels + 1) = M_OR_FT(3,10);
+		decostoplevels[1] = m_or_ft(3,10);
```

Comment:
Should this have a space after the comma?

Reply:
Fixed here and elsewhere.